packages/replay-next/src/contexts/FocusContext.tsx

```diff
@@ -132,5 +134,9 @@ export function FocusContextRoot({ children }: PropsWithChildren<{}>) {
     [deferredRange, isTransitionPending, range, updateFocusRange]
   );
 
-  return <FocusContext.Provider value={focusContext}>{children}</FocusContext.Provider>;
+  return (
+    <FocusContext.Provider value={focusContext}>
+      {focusRangeInitialized ? children : null}
```

Comment:
> It will delay showing anything in the production app

Does it? I thought that `FocusContextRoot` is only used in `replay-next` but the production app uses `FocusContextReduxAdapter`. Otherwise I'd agree that we need to find a way to fix this without affecting prod.

Reply:
Ah, right. `FocusContextReduxAdapter` replaces `FocusContextRoot` in our production app rather than decorating it.

This change still feels pretty brittle to me..

I guess this is a problem **now** because of the recent backend change that throws "_The focus range was changed while the command was in progress_" ? I wonder how else that change is going to bite us.

Reply:
> I guess this is a problem now because of the recent backend change that throws "The focus range was changed while the command was in progress" ?

Yes

> I wonder how else that change is going to bite us.

Right now I think it can bite us in a lot of places. But we usually pass our own `range` parameter in commands that need a range and in that case the current focus range is irrelevant, so this error shouldn't be thrown. Filed BAC-3030 for that and when that is fixed I think it shouldn't be a problem anymore.

Reply:
Should we land this for now and back it out when BAC-3030 is fixed?

Reply:
BAC-3030 looks good. I'd feel better about landing this temporarily if we knew BAC-3030 would be accepted.

Something about the way this all relies on mount sequence and timing etc really nags at me about relying on it though. Feels really hacky.

Reply:
I think BAC-3030 really has to be fixed - the current behavior makes it unnecessarily complicated to write a UI that works reliably.

> Feels really hacky.

Yes it does. But I can't think of a cleaner way to work around BAC-3030 and since (IMHO) that issue really needs to be addressed in the backend I'd vote for landing this temporarily.